File: CreditTask1-Attempt1/CT1/CT1/Question.cs
// FileName: Question.cs
// Author: Lewis Brockman-Horsley
// Email: 101533222@students.swinburne.edu.au
// Date: 24/10/2019
using System;
using System.Collections.Generic;
namespace CT1
{
    public class Question
    {

        private int _questionID;
        private string _question;
        private List<string> _choices;
        private int _answer;

        public int QuestionID { get => _questionID; set => _questionID = value; }
        public int Answer { get => _answer; set => _answer = value; }

        public Question(int id, string question, List<string> choices, int answer)
        {
            _questionID = id;
            _question = question;
            _choices = choices;
            _answer = answer;
        }

        public void CreateChoice(String choice)
        {
            _choices.Add(choice);
        }

        public void EditChoice(int i, string choice)
        {
            _choices[i] = choice;
        }

        public void DeleteChoice(int i)
        {
            _choices.RemoveAt(i);
        }

        public void SetAnswer(int a)
        {
            _answer = a;
        }

        public void Display()
        {
            Console.WriteLine(_question + "\n");
            int i = 0;
            foreach (string choice in _choices)
            {
                Console.WriteLine(i++ + ": " + choice);
            }
            Console.WriteLine("Enter Answer(number): ");
        }
    }
}

File: Week07/CT2/Question.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Question.cpp
 * Author: lewis
 * 
 * Created on 6 November 2019, 2:41 PM
 */


#include "Question.h"
#include <iostream>

/**
 * Constructor
 * @param id unique int for question ID
 * @param question string of the actual question
 * @param choices possible answers for the multiple choices
 * @param answer int index of the answer from choices
 */
Question::Question(int id, std::string question, std::vector<std::string> choices, int answer) {
    _questionID = id;
    _question = question;
    _choices = choices;
    _answer = answer;
}

Question::Question(const Question& orig) {
}

Question::~Question() {
}

/**
 * Create a new choice for this question
 * @param choice
 */
void Question::CreateChoice(std::string choice){
    _choices.push_back(choice);
}

/**
 * Replaces the choice at a given index with the new string
 * @param i
 * @param choice
 */
void Question::EditChoice(int i, std::string choice){
    _choices[i] = choice;
}

/**
 * Deletes a choice given an index
 * @param i index of choice to delete
 */
void Question::DeleteChoice(int i){
    _choices.erase(_choices.begin() + i);
}

/**
 * Updates the answer index
 * @param a index of new answer
 */
void Question::setAnswer(int a){
    _answer = a;
}

/**
 * gets the answer index
 * @return int of answer index
 */
int Question::getAnswer() {
    return _answer;
}

/**
 * get the question ID
 * @return question ID
 */
int Question::getId() {
    return _questionID;
}

/**
 * get the list of choices, used for testing
 * @return the list of choices
 */
std::vector<std::string> Question::getChoices() {
    return _choices;
}

/**
 * gets the number of choices of this question, used for testing
 * @return number of choices
 */
int Question::getNumberOfChoices() {
    return _choices.size();
}

/**
 * Displays the Question on the console
 */
void Question::Display(){
    std::cout << _question << std::endl;
    for (int i = 0; i < _choices.size(); i++){
        std::cout << i << ": " << _choices[i] << std::endl;
    }
    std::cout << "Enter Answer(number): " << std::endl;
}